src/qt/test/apptests.h

```diff
@@ -0,0 +1,39 @@
+#ifndef BITCOIN_QT_TEST_APPTESTS_H
+#define BITCOIN_QT_TEST_APPTESTS_H
+
+#include <QObject>
+
+class BitcoinApplication;
+class BitcoinGUI;
+class RPCConsole;
+
+class AppTests : public QObject
+{
+    Q_OBJECT
+public:
+    AppTests(BitcoinApplication& app) : m_app(app) {}
+
+private Q_SLOTS:
+    void appTests();
+    void guiTests(BitcoinGUI* window);
+    void consoleTests(RPCConsole* console);
+
+private:
+    //! Increment number of pending callbacks.
+    void expectCallback() { ++m_callbacks; }
+
+    //! RAII helper to decrement number of pending callbacks.
+    struct HandleCallback
+    {
+        AppTests& m_app_tests;
+        ~HandleCallback();
+    };
+
+    //! Bitcoin application.
+    BitcoinApplication& m_app;
+
+    //! Number of expected callbacks pending.
+    int m_callbacks = 0;
```

Comment:
`QStack<QString> m_callbacks;`. See above.

Reply:
Responded above, would have to be a multiset instead of a stack.